app/classifier/index.cjsx

```diff
@@ -34,13 +38,35 @@ Classifier = React.createClass
   getInitialState: ->
     subjectLoading: false
     backButtonWarning: false
+    renderIntervention: false
+
+  disableIntervention: ->
+    @setState renderIntervention: false
+
+  enableIntervention: ->
+    @context.experimentsClient.logExperimentState @context.geordi, @context.interventionMonitor?.latestFromSugar, "interventionDetected"
+    @setState renderIntervention: true
 
   componentDidMount: ->
+    @context.experimentsClient.startOrResumeExperiment @context
+    @setState renderIntervention: @context.interventionMonitor?.shouldShowIntervention()
+    @context.interventionMonitor.on 'interventionRequested', @enableIntervention
+    @context.interventionMonitor.on 'classificationTaskRequested', @disableIntervention
     @loadSubject @props.subject
     @prepareToClassify @props.classification
     {workflow, project, preferences, user} = @props
     Tutorial.startIfNecessary {workflow, user, preferences}
 
+  reCheckIfInterventionNeeded: ->
+    # Sometimes, the intervention request arrives at a time when this component is not able to receive that
```

Comment:
I don't think this comments apply anymore as you aren't checking if the component is mounted.